packages/react/src/internal/useResizeObserver.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { useRef, useState, useLayoutEffect, useEffect } from 'react';
+
+export const useResizeObserver = ({ ref, onResize }) => {
+  const [width, setWidth] = useState(-1);
+  const [height, setHeight] = useState(-1);
+  const entriesToHandle = useRef<ResizeObserverEntry[] | null>(null);
+  const cb = useRef(onResize);
+
+  useEffect(() => {
+    // ref for onResize removes it as dependency from useLayoutEffect
+    // This significantly reduces repeated calls if a function is redefined on every
+    // render
+    cb.current = onResize;
+  }, [onResize]);
+
+  useEffect(() => {
+    const getInitialSize = () => {
+      if (ref.current) {
+        const refComputedStyle = window.getComputedStyle(ref.current);
+
+        const initialWidth =
+          (ref.current?.offsetWidth || 0) -
+          (typeof refComputedStyle?.paddingLeft === 'string' &&
+          refComputedStyle?.paddingLeft.length
+            ? parseFloat(refComputedStyle?.paddingLeft)
+            : 0) -
+          (typeof refComputedStyle?.paddingRight === 'string' &&
+          refComputedStyle?.paddingRight.length
+            ? parseFloat(refComputedStyle?.paddingRight)
+            : 0);
+
+        const initialHeight =
+          (ref.current?.offsetHeight || 0) -
+          (typeof refComputedStyle?.paddingTop === 'string' &&
+          refComputedStyle?.paddingTop.length
+            ? parseFloat(refComputedStyle?.paddingTop)
+            : 0) -
+          (typeof refComputedStyle?.paddingLeft === 'string' &&
+          refComputedStyle?.paddingLeft.length
+            ? parseFloat(refComputedStyle?.paddingLeft)
```

Comment:
Should `paddingLeft` be `paddingBottom` instead? A height is being calculated. If it should be `paddingBottom`, should a test be failing given it's not?